src/box/lua/config/configdata.lua

```diff
@@ -888,6 +888,55 @@ local function validate_replicaset_names_are_unique(cconfig)
     end
 end
 
+local function validate_instance_names_are_unique(cconfig)
+    local instance_groups = {}
+    local instance_replicasets = {}
+
+    for group_name, group in pairs(cconfig.groups) do
+        for replicaset_name, replicaset in pairs(group.replicasets) do
+            for instance_name, _ in pairs(replicaset.instances) do
+                local dup_group_name = instance_groups[instance_name]
+                local dup_replicaset_name =
+                    instance_replicasets[instance_name]
+
+                -- Duplicating instance name is found within the
+                -- same replicaset.
+                if replicaset_name == dup_replicaset_name then
+                    assert(group_name == dup_group_name)
+
+                    error(('found two replicasets with the same name %q in ' ..
+                           'the group %q.')
+                          :format(replicaset_name, group_name), 0)
+                end
```

Comment:
The error message contradicts with the comment above the code block.

Reply:
You're right. Thank you for noticing. Fixed.